### ISSUE TITLE
NAS Synology and Syncthing - Syncthing not starting

### ISSUE THREAD

**Issue by safersurfing** [SynoCommunity/spksrc#2176]
I have a NAS Synology DS715 and Syncthing 0.12.8-7 installed. My issue may relate to the above comments.

I start Syncthing (run syncthing via the package center, but it stops straight away. The only logs I have found are following.
Any idea?

2016/03/04 06:33:37 start syncthing: begin to start version 0.12.8-7
2016/03/04 06:33:37 start syncthing: start version 0.12.8-7 successfully, result 0

Mar 4 06:33:42 DiskStation php: Failed to delete keys [F3ECC2A519CA01C85D04EDA6D0C2F18694A0B88E].
Mar 4 06:33:42 DiskStation php: DeleteOtherKeyServer failed.
##

**Comment by Dr-Bean** [comment]
Missing some information here, e.g. DSM version. Please follow instructions here: https://github.com/SynoCommunity/spksrc/blob/master/CONTRIBUTING.md#issue-content, and add this please.

Try and start the package manually (log in via SSH as root): https://github.com/SynoCommunity/spksrc/blob/master/spk/syncthing/src/dsm-control.sh#L17:
```
start-stop-daemon -b -o -c syncthing -S -u syncthing -x env HOME=/usr/local/syncthing/var/ /usr/local/syncthing/bin/syncthing -- --home /usr/local/syncthing/var/
```

**Comment by safersurfing** [comment]
OK, here you go:

Model name
DS715
CPU
ANNAPURNALABS Alpine AL314
CPU clock rate
1.4 GHz
CPU cores
4
Total physical memory
2048 MB
DSM version
DSM 5.2-5644 Update 5

I canont find logs and do not know how to set them up.

I tried to enter what you wrote see below and got following response:
sh start-stop-daemon -b -o -c syncthing -S -u syncthing -x env HOME=/usr/local/syncthing/var/ /usr/local/syncthing/bin/syncthing -- --home /usr/local/syncthing/var/

start-stop-daemon: line 1: syntax error: unexpected word (expecting ")")

Is there a chance you could login life via teamviewer or anydesk onto my NAS?

**Comment by cytec** [comment]
@safersurfing spamming all syncthing issues won't get you a solution any faster...

**Comment by safersurfing** [comment]
sorry

**Comment by Dr-Bean** [comment]
Two things:
- If you have made any changes to the package, e.g. the user Syncthing runs as, such as described in that link you referred to, you're on your own. There's no way we can support altered packages.
- Note that Syncthing's webui takes a while to start (probably up to a couple of minutes). In other words: after starting the package (either via Package Center or manually), I would expect a Syncthing process to be running. If that's the case, just wait longer.

You can verify that with something like `ps -www | grep syncthing`, which should return a running process or two. It looks like this:
```
NAS> /var/packages/syncthing/scripts/start-stop-status stop 
Stopping Syncthing ...
stopped /usr/local/syncthing/bin/syncthing (pid 25956 25951)
NAS> ps -www | grep syncthing
25982 root      4188 S    grep syncthing
NAS> /var/packages/syncthing/scripts/start-stop-status start
Starting Syncthing ...
NAS> ps -www | grep syncthing
25951 syncthin  779m S    /usr/local/syncthing/bin/syncthing --home /usr/local/syncthing/var/
25956 syncthin  779m S    /usr/local/syncthing/bin/syncthing --home /usr/local/syncthing/var/
25971 root      4188 R    grep syncthing
```

**Comment by safersurfing** [comment]
No changes to the package. Is installed as provided (Alpine).
Waiting more than 10 minutes, however what I found was following log:

Found following ... DiskStation> /usr/local/syncthing/ /usr/local/syncthing/app/ /usr/local/syncthing/bin/ /usr/local/syncthing/var/ DiskStation> /usr/local/syncthing/bin/syncthing unexpected fault address 0x492cf4 fatal error: fault [signal 0xb code=0x2 addr=0x492cf4 pc=0x492cf4]

goroutine 1 [running, locked to thread]: runtime.gothrow(0x5b4278, 0x5) /spksrc/native/go/work-native/go/src/runtime/panic.go:503 +0x84 fp=0x10c1b61c sp=0x10c1b610 runtime.sigpanic() /spksrc/native/go/work-native/go/src/runtime/sigpanic_unix.go:29 +0x2a4 fp=0x10c1b644 sp=0x10c1b61c github.com/golang/snappy.init() ?:0 fp=0x10c1b648 sp=0x10c1b648 github.com/syndtr/goleveldb/leveldb.init() /spksrc/spk/syncthing/work-alpine-5.2/syncthing/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/version.go:457 +0x70 fp=0x10c1b680 sp=0x10c1b648 github.com/syndtr/goleveldb/leveldb.init() /spksrc/spk/syncthing/work-alpine-5.2/syncthing/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/version.go:457 +0x70 fp=0x10c1b6b8 sp=0x10c1b680 github.com/syncthing/syncthing/lib/db.init() /spksrc/native/go/work-native/go/src/github.com/syncthing/syncthing/lib/db/virtualmtime.go:76 +0x8c fp=0x10c1b6e8 sp=0x10c1b6b8

goroutine 2 runnable: runtime.forcegchelper() /spksrc/native/go/work-native/go/src/runtime/proc.go:90 runtime.goexit() /spksrc/native/go/work-native/go/src/runtime/asm_arm.s:1322 +0x4

goroutine 3 runnable: runtime.bgsweep() /spksrc/native/go/work-native/go/src/runtime/mgc0.go:82 runtime.goexit() /spksrc/native/go/work-native/go/src/runtime/asm_arm.s:1322 +0x4

goroutine 4 runnable: runtime.runfinq() /spksrc/native/go/work-native/go/src/runtime/malloc.go:712 runtime.goexit() /spksrc/native/go/work-native/go/src/runtime/asm_arm.s:1322 +0x4

goroutine 5 /spksrc/native/go/work-native/go/src/github.com/syncthing/syncthing/lib/dialer/internal.go:46 created by github.com/syncthing/syncthing/lib/dialer.init·1 /spksrc/native/go/work-native/go/src/github.com/syncthing/syncthing/lib/dialer/internal.go:49 +0x260

goroutine 6 /spksrc/native/go/work-native/go/src/os/signal/signal_unix.go:19 created by os/signal.init·1 /spksrc/native/go/work-native/go/src/os/signal/signal_unix.go:27 +0x40

**Comment by safersurfing** [comment]
Tried Dr-Bean suggestion, but only get 25982 root      4188 S    grep syncthing returned, no matter if stop or start entered.
It looks like syncthing is actually not running. The log see above is shown.

**Comment by safersurfing** [comment]
Anybody have an idea what the issue could be here?

**Comment by safersurfing** [comment]
Really stuck here ... it looks like syncthing does not run on DS715 Synology (Alpine).

**Comment by Dr-Bean** [comment]
ping @cytec Maybe update Go to whatever version Syncthing uses (1.5 last time I checked, could be even newer now)

@safersurfing Here's an alternative then, at least unless we can figure out what causes the package to fail on Alpine.
Get Syncthing's build here: https://github.com/syncthing/syncthing/releases/download/v0.12.21/syncthing-linux-arm-v0.12.21.tar.gz. Extract the `syncthing` binary, and put that in `/usr/local/syncthing/bin/`, replacing/overwriting the existing file.

**Comment by cytec** [comment]
yeah... was thinking on updateing the go stuff for a while now... starting with 1.5 cross compile became a lot easier to handle and we may be able to produce arm5 and arm7 bins ...

**Comment by dundeemt** [comment]
Experiencing the same here - Installed the update yesterday, added a device today, which required a restart -- then just silence.   Attempted Stop/Start through web interface.  It says, stopped and now says running except the web interface is now "Connection Refused."

Model: DS410j
CPU MARVELL Kirkwood 88F6281
CPU clock rate 800 MHz
CPU cores 1
Total physical memory 128 MB
DSM version DSM 5.2-5644 Update 5
System time 2016-04-10 13:28:32
Uptime 14 day(s) 22 hour(s) 7 minute(s) 39 second(s)
Thermal status Normal

Is there a way to roll back?  Has been working fine up to now.

**Comment by cytec** [comment]
@Dr-Bean so... i've just looked into this and it seems like since go 1.5 you'll need 2 versions of go to build the cross compiler... basically you need to have go >= 1.4 installed to build the 1.5+ binaries. How are we gonna handle this? donwload and build 1.4 in the same Makefile via `CONFIGURE_TARGET` or better create a new `native/go1.6` package and depend on the "old" `native/go`?

**Comment by Dr-Bean** [comment]
Hm...there doesn't seem to be a way around it indeed. The debian packages in stable are all too old to use for bootstrapping (both `gccgo` and `golang-go`), and getting them from testing or jessie-backports is a hassle.

Are there any reasons/upsides for us to provide different versions of `go`? E.g a `go-1.5` alongside a `go-1.6`?
If we can get away with always using the latest version of `go`, maybe do it the other way around as you suggested: set up `native/go` and `native/go-1.4`. Bootstrap `native/go` with `native/go-1.4`, and continue to use `native/go` to depend upon from Syncthing or other cross packages.

**Comment by cytec** [comment]
i'd go with your suggestion to use latest go version and use go-1.4 for bootstrapping... maybe that'll change in the future and we can get rid of `native/go-1.4` again at some point...

**Comment by scyto** [comment]
I have 1815+ with DSM 6 update 3 and syncthing fails to start from the UI, no logs in the UI.  I am unclear where to go / what to do after i SSH in?  I tried start-stop-daemon command above but it said command not found.

**Comment by erikvanoosten** [comment]
I had similar problems. Syncthing wouldn't start up after a reboot of my synology.

Syncthing started working once I realized that:
* no `syncthing` user was created
* `/usr/local/syncthing` has rights that are all over the place, not all them can be read by the `syncthing` user

As a 'fix' (not sure it is a fix) I did the following:
* Created user `syncthing`, put it in the `sc-syncthing` group and gave it access to the syncthing share
* Logged in via ssh and executed `sudo chown -R syncthing:users /usr/local/syncthing`

Now syncthing started fine. I wonder what will happen on the next reboot.
This issue might be related to #2140.

**Comment by Jamesits** [comment]
Have the same issue on DS1817+, DSM 6.1.3-15152 Update 1. Syncthing stopped and refused to start after a reboot.
```
$ /var/packages/syncthing/scripts/start-stop-status start
start-stop-daemon: unknown user syncthing
Starting Syncthing ...
start-stop-daemon: unknown user syncthing
```

**Comment by cinghialino** [comment]
@Jamesits did you fixed it? I noticed that there is a new update of the DSM, maybe will start back..

**Comment by Jamesits** [comment]
@cinghialino I've applied the newest update, but it seems to be only firmware updates:

> Fixed Issues
> 1. BIOS update to improve system stability. Applied models: DS1515+, DS1517+, DS1815+, DS1817+, DS2415+, DS415+, RS2416+, RS2416RP+, RS815+, RS815RP+.
> 2. Enhanced USB 2.0 compatibility for DS415+, RS2416+, and RS2416RP+.

I gave up running this version of Syncthing on DSM... Waiting for a clear fix. Reinstalling wipes all its config and I need a lot of time setting it up again. I'm using VM Manager to run a Ubuntu in DSM, install Syncthing on that OS and mount shared folders over CIFS. (This is not stable either but at least no serious bugs)

**Comment by cinghialino** [comment]
@Jamesits I agree we synology users need a easy package.

**Comment by shavenewok** [comment]
Hi All

I followed @erikvanoosten instructions and still couldn't get  it to work.

I found i didn't have the rights to access /usr/local/syncthing/var 

Once i ran sudo chown -R syncthing:users /usr/local/syncthing/var i was able to start Syncthing from the package centre. I also tested with a reboot and Syncthing auto started after the reboot

**Comment by erikvanoosten** [comment]
Unfortunately it soon stopped working for me also. I gave up and now run syncthing on a Ubuntu machine.

**Comment by jamesd** [comment]
On my system, `/usr/local/syncthing` is a symlink to where the appstore puts syncthing and the chmod did not follow through the link.

I changed the chmod in @erikvanoosten's intructions to:

    sudo chown -R syncthing:root /usr/local/syncthing "$(readlink -f /usr/local/syncthing)"

syncthing has been stable through reboots for me for awhile now.

**Comment by KingDuckZ** [comment]
Same problem on DSM 6.2-23739 Update 2
I logged into ssh and:

```
admin@DiskStation:/usr/local/syncthing/var$ /var/packages/syncthing/scripts/start-stop-status stop
/var/packages/syncthing/scripts/service-setup: line 6: [: : integer expression expected
/var/packages/syncthing/scripts/start-stop-status: line 43: start-stop-daemon: command not found
 is not running

admin@DiskStation:/usr/local/syncthing/var$ /var/packages/syncthing/scripts/start-stop-status status
/var/packages/syncthing/scripts/service-setup: line 6: [: : integer expression expected
/var/packages/syncthing/scripts/start-stop-status: line 43: start-stop-daemon: command not found
 is not running

admin@DiskStation:/usr/local/syncthing/var$ /var/packages/syncthing/scripts/start-stop-status start
-sh: /var/packages/syncthing/scripts/start-stop-status: No such file or directory
```

Otherwise I seem to have the sc-syncthing user set correctly in /etc/passwd and file permissions look ok to me. syncthing version available at this time is  v0.14.49-15 https://synocommunity.com/package/syncthing

**Comment by KingDuckZ** [comment]
acolomb it doesn't start from the gui either, that's why I had to try over ssh.
